parquet-hadoop/src/main/java/org/apache/parquet/hadoop/metadata/BlockMetaData.java

```diff
@@ -46,10 +46,17 @@ public BlockMetaData() {
 
   // Reader side (get parameters from RowGroup structure)
   public BlockMetaData(long fileOffset, long totalCompressedSize) {
-    this.startingPosition = fileOffset;
-    startingPositionSet = true;
-    this.totalCompressedSize = totalCompressedSize;
-    totalCompressedSizeSet = true;
+    // fileOffset is optional, 0 means fileOffset is absent
```

Comment:
In the current mr code, it is called from one place only, on the reader path,
 ParquetMetadataConverter.fromParquetMetadata
and never with 0 values. 

Are you adding these checks for another mr code version (eg with changes for Presto?). If yes, please let me know, and please send a commit for that. I think its best to close the current pr#5 (still shows changes in 3 files..), and open a new pr for the BlockMetaData constructor checks.